go-controller/pkg/ovn/loadbalancer/loadbalancer.go

```diff
@@ -36,41 +36,58 @@ import (
 // It is assumed that names are meaningful and somewhat stable, to minimize churn. This
 // function doesn't work with Load_Balancers without a name.
 func EnsureLBs(nbClient libovsdbclient.Client, externalIDs map[string]string, LBs []LB) error {
-	lbCache, err := GetLBCache(nbClient)
+	existing := libovsdbops.FindLoadBalancersByExternalIDs(nbClient, externalIDs)
+	existingByName := make(map[string]nbdb.LoadBalancer, len(existing))
+	toDelete := sets.NewString()
+
+	// gather some data
+	lbToSwitchNames, switchNamesToUUID, err := listSwitches(nbClient)
 	if err != nil {
-		return fmt.Errorf("failed initialize LBcache: %w", err)
+		return err
+	}
+	lbToRouterNames, routerNamesToUUID, err := listRouters(nbClient)
+	if err != nil {
+		return err
+	}
+	lbToGroupNames, groupNamesToUUID, err := listGroups(nbClient)
+	if err != nil {
+		return err
 	}
-
-	existing := lbCache.Find(externalIDs)
-	existingByName := make(map[string]*CachedLB, len(existing))
-	toDelete := sets.NewString()
 
 	for _, lb := range existing {
+		// collision - somehow things didn't come together how we expected
+		// rare but we have seen it happen, see https://bugzilla.redhat.com/show_bug.cgi?id=2042001
+		if _, ok := existingByName[lb.Name]; ok {
+			klog.V(2).Infof("Name collision for load balancer %s: deleting both and re-creating")
+			toDelete.Insert(lb.UUID)
+			delete(existingByName, lb.Name)
```

Comment:
What if we have triplicates? Would it be better to keep the first one and update instead of deleting both?

Reply:
On another note, I have the feeling that we are not uncovering the original issue why this happens and that we might end up masquerading other issues with this types of workarounds.

Reply:
The retries are news to me :/ I thought we were blocking the transaction on reconnect. 

Definitely, one of the workers of the service controller could be held up in that retry while other worker could fit in an update for the same service in the meantime.

Reply:
And I guess it would not matter whether we have a retry or we are blocking, that's just a way of saying it's happening at a different level.

Reply:
So @trozet let me know that the same service will not be processed by the work queue more than once, so that's not it.

Reply:
I could see TransactWithRetry potentially creating something, even though the client disconnects and then we retry and recreate that thing again. The other workaround im writing up I think would catch this. We could change that log level on teh log message that would hint to us this is what is happening:
klog.V(5).Infof("Unable to execute transaction: %+v. Client is disconnected, will retry...", ops)

However, the workqueues always distribute the same key to the same worker go routine. So I doubt there is any issue with workers clashing on creating things.

Reply:
If we have triplicates, then we just adopt that last one. I could flag that name as "tainted" instead - probably a good idea.

Reply:
Ok, I guess it seemed unintentional or random to me that if we had duplicates we would remove both but if we had triplicates we would leave the last one around, and so on...
Also, I was not sure we could assume there were 100% dups so wanted to trigger the discussion of whether the appropriate thing to do would be to remove all regardless of duplicate, triplicate, etc..
Anyway, the duplicate discussion might be irrelevant now if the other fixes we have for that  end up working fine.